### ISSUE TITLE
[Documentation] Collab notebook for GPT-J Batch Prediction Ray AIR example fails with Ray not installed, also some broken links

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#33658]
### Description

[This documentation example](https://docs.ray.io/en/master/ray-air/examples/gptj_batch_prediction.html) for GPT-J Batch Prediction with Ray AIR links to a Collab notebook.

<img width="903" alt="Screen Shot 2023-03-23 at 7 17 59 PM" src="https://user-images.githubusercontent.com/249517/227407483-4490e0cd-27eb-41f7-a91e-ec228940569d.png">

### Ray not installed

Running the first few cells in Collab leads to the following failure

<img width="923" alt="Screen Shot 2023-03-23 at 7 18 35 PM" src="https://user-images.githubusercontent.com/249517/227407565-d35fba3b-7e8c-44d6-8c46-7821c1365712.png">

This may be true of other examples as well, I didn't check.

### Broken links

Also, 2 of the first 4 links appear to be broken.

<img width="1042" alt="Screen Shot 2023-03-23 at 7 20 23 PM" src="https://user-images.githubusercontent.com/249517/227407812-48f7f325-2067-4851-952e-ac74899cb109.png">

**Comment by richardliaw** [comment]
For all notebooks, the colab conversion basically doesn't work if you have relative links, etc.

cc @maxpumperla

**Comment by Yard1** [comment]
As a short term fix, I'll add cells with `pip install` to the LLM notebooks

**Comment by Yard1** [comment]
Actually, @robertnishihara, those notebooks won't run in colab as they need more memory than what colab provides. Not sure if we can disable automatic conversion for them? @maxpumperla

**Comment by maxpumperla** [comment]
@Yard1 what do you mean by conversion? this is already a notebook, right? do we want to deactivate the colab button?

btw, the relative links not working can't be avoided, if we open a notebook outside of the sphinx ecosystem. using absolute URLs is not a great fix, though.

**Comment by Yard1** [comment]
Yes, deactivate the colab button for specific examples.

BTW. would it be possible to fix the link issue by replacing the links with absolute versions when the notebook is converted/downloaded?

**Comment by maxpumperla** [comment]
yeah, I think we can to this (deactivate the button). let me check, I think you can configure this in the frontmatter. Also, if something is not executable, the easiest way is to just convert ipynb to md (or rst), then it won't be seen as notebook by sphinx.

not really, as there is no conversion. it's literally just the source notebook with a prefix url for colab.

**Comment by richardliaw** [comment]
I think Antoni is asking, what would it take to engineer a script and
generate a sanitized version that works for colab

**Comment by Yard1** [comment]
^ this

**Comment by maxpumperla** [comment]
ah, got it. that's why I asked what is meant by conversion. should be doable

**Comment by richardliaw** [comment]
Max, could you help label a priority here / provide a suggestion as to how to proceed? Should we just disable the colab button or invest in building the infra?

**Comment by maxpumperla** [comment]
@richardliaw I think https://github.com/ray-project/ray/pull/33912 (or something similar we all agree on) would resolve the "Colab doesn't work" part.

When it comes to the relative links, without further investigation of how relevant this is (I assume very, very few users ever use the Colab buttons, and 5% of those ever think about those links), this should be quite low on our list of things to tackle.

**Comment by amogkam** [comment]
Let's confirm whether this example can work on Colab or not. The example says it requires at least 16 GB GPU memory. Colab has Tesla T4 GPUs which have 16 GB memory.

If this example does not work, then let's update the message in the example and remove the option to run on Colab.

**Comment by richardliaw** [comment]
Can you actually disable colab per example? I think it is a site-wide
option.

**Comment by amogkam** [comment]
Ah I see @Yard1, the issue is there is not enough CPU RAM to load the model (only ~11 GB is available on colab free tier).

**Comment by amogkam** [comment]
> I assume very, very few users ever use the Colab buttons, and 5% of those ever think about or care about those broken links

If this is the case, then let's just remove the colab button?

**Comment by dstrodtman** [comment]
Closing as obsolete. Ray AIR was deprecated in 2023, and the APIs referenced here have moved into Ray Train (typically with reshaped surface area). If you're hitting a similar gap in the current Train docs, please open a fresh issue with a link to the page.

Context: docs-team triage of the OSS docs issue backlog.
